data_structures/src/radon_error.rs

```diff
@@ -52,11 +52,17 @@ pub enum RadonErrors {
     TallyExecution = 0x53,
     /// Invalid reveal serialization (malformed reveals are converted to this value)
     MalformedReveal = 0x60,
+    /// Invalid data request serialization (malformed data requests are converted to this value)
+    MalformedRequest = 0x61,
     // Access errors
     /// Tried to access a value from an index using an index that is out of bounds
     ArrayIndexOutOfBounds = 0x70,
     /// Tried to access a value from a map using a key that does not exist
     MapKeyNotFound = 0x71,
+    // This error it is not really an error. It is need to notify to an external contract that some
+    // of the parameters specified in the request, exceed the capabilities that a bridge node is willing to offer
+    /// Wit requirements for the data request exceed the capabilities that a bridge node is willing to offer
+    ExpensiveRequest = 0x80,
```

Comment:
Aha — it looks like there are multiple reasons why a relayer may refuse to submit an otherwise perfectly valid request.

Maybe we can reflect on the thing that is common to all those reasons: the request is lacking enough incentives to compel the relayer to submit the transaction and put their wits at stake.

```suggestion
    /// The request is rejected on the grounds that it may cause the submitter, the
    /// committers or the revealers to spend or stake an amount of value that is
    /// unjustifiably high when compared with the reward they will be getting. 
    PoorIncentives = 0x80,
```

Reply:
I'm now wondering if this is a RadonError that we may want to allow in commits, reveals and tallies.

Currently our take would be to refrain from committing, and let the request time out with a tally stating that it didn't get enough commits.

Of course, going down the route of allowing this error in commits, reveals and tallies would mean that all nodes would be required to enact the same criteria for considering whether a request "is worth it". That's just the opposite to our current take, in which we preferred to let each node decide how much they want to put their wits into risk.

Reply:
I think this error is an inter-client communication error that you commented before, because it only affect when there are bridges nodes that refuse to include it in Witnet. In the Witnet protocol, the person who wants to include a request and the person who finally include it is the same, and there are the witnesses who decides to solve or not, but it exist a RadonError for that (InsufficientCommits)

Reply:
`PoorIncentives` is too broad, I prefer to split this error into the different cases:

* Too expensive for the relayer
* Collateral too high
* Reward too low
* Commit/reveal fees too low

Reply:
I think that collateral too high, reward too low and commit/reveal fees too low should be considered by witnesses, not by the bridge node operator, because to that problems, there already are a RadonError::InsufficientCommits in case that witnesses or miners decide not to solve that request